src/infrastructure/fileUploader.repository.ts

```diff
@@ -0,0 +1,58 @@
+import type NotesApiTransport from './transport/notes-api';
+import type { Note } from '@/domain/entities/Note';
+import mime from 'mime';
+import type FileUploaderRepositoryInterface from '@/domain/fileUploader.repository.interface';
+
+/**
+ * Facade for interacting with files
+ */
+export default class FileUploaderRepository implements FileUploaderRepositoryInterface {
+  /**
+   * Transport instance
+   */
+  private readonly transport: NotesApiTransport;
+
+  /**
+   * Repository constructor
+   * @param notesApiTransport - notes api transport instance
+   */
+  constructor(notesApiTransport: NotesApiTransport) {
+    this.transport = notesApiTransport;
+  }
+
+  /**
+   * Upload file data
+   * @param noteId - note indentifier to make attachment
+   * @param fileData - file binary data
+   * @param fileName - file name
```

Comment:
describe a case, when fileName can be omitted

Reply:
removed